### PR TITLE
Implement Nuget plugin fetcher

### PR DESCRIPTION
**Pull request by SakuraIsayeki**

### CODE REVIEW

File: src/YumeChan.Core/Services/Plugins/NugetPluginsFetcher.cs [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+			if (File.Exists(Path.Combine(Assembly.GetExecutingAssembly().Location, dep.Id + ".dll")))
+			{
+				return true;
+			}
+		}
+		
+		return runtimeLib is not null;
+	}
+	
+	private static async Task FlattenDownloadedPackageToDirectoryStructureAsync(DirectoryInfo downloadFolder, DirectoryInfo targetFolder, CancellationToken ct = default)
```
Comment (tebeco) [inline src/YumeChan.Core/Services/Plugins/NugetPluginsFetcher.cs:293]:
```suggestion
	private static Task FlattenDownloadedPackageToDirectoryStructureAsync(DirectoryInfo downloadFolder, DirectoryInfo targetFolder, CancellationToken ct = default)
```
🦤

image

File: src/YumeChan.Core/YumeCore.cs [hunk 2]
```diff
@@ -135,8 +149,8 @@ private string GetBotToken()
 				else
 				{
 					ApplicationException e = new("No Bot Token supplied.");
-					Logger.LogCritical(e, "No Bot Token was found in \"coreconfig.json\" Config File, and Environment Variables \"{EnvVar}\" from relevant targets are empty. " +
-											$"\nPlease set a Bot Token before launching the Bot.", envVarName);
+					Logger.LogCritical(e, "No Bot Token was found in \"coreconfig.json\" Config File, and Environment Variables \"{EnvVar}\" from relevant targets are empty. " + $"\nPlease set a Bot Token before launching the Bot.", envVarName);
```
Comment (tebeco) [inline src/YumeChan.Core/YumeCore.cs:152]: '\n` is log is debattable

File: src/YumeChan.Core/Services/Plugins/NugetPluginsFetcher.cs (hunk 1 above)
Comment (SakuraIsayeki) [reply src/YumeChan.Core/Services/Plugins/NugetPluginsFetcher.cs]: Fixed in 7ce54622ab8ef62bd4b617e4fbd9c175342ff156.

File: src/YumeChan.Core/YumeCore.cs (hunk 2 above)
Comment (SakuraIsayeki) [reply src/YumeChan.Core/YumeCore.cs]: Fixed in 7ce54622ab8ef62bd4b617e4fbd9c175342ff156.

**Review by CalebABG — APPROVED** [review]
LGTM 🚀

**Pull request merged**